torch_geometric/nn/models/graph_cross_network.py

```diff
@@ -0,0 +1,588 @@
+
+    def forward(self, graph: Batch, feat: Tensor, select_idx: Tensor,
+                non_select_idx: Optional[Tensor] = None,
+                scores: Optional[Tensor] = None, pool_graph=False):
+        r"""
+        Description
+        -----------
+        Perform graph pooling.
+
+        Parameters
+        ----------
+        graph : torch_geometric.data.Batch
+            The input graph
+        feat : torch.Tensor
+            The input node feature
+        select_idx : torch.Tensor
+            The index in fine graph of node from
+            coarse graph, this is obtained from
+            previous graph pooling layers.
+        non_select_idx : torch.Tensor, optional
+            The index that not included in output graph.
+            default: :obj:`None`
+        scores : torch.Tensor, optional
+            Scores for nodes used for pooling and scaling.
+            default: :obj:`None`
+        pool_graph : bool, optional
+            Whether perform graph pooling on graph topology.
+            default: :obj:`False`
+        """
+        if self.use_gcn:
+            feat = self.down_sample_gcn(feat, graph.edge_index)
+
+        feat = feat[select_idx]
+        if scores is not None:
+            feat = feat * scores.unsqueeze(-1)
+
+        if pool_graph:
+            sub_graph = graph.clone()
+            sub_graph.num_nodes = select_idx.shape[0]
+            sub_graph.x = sub_graph.x[select_idx]
+            sub_graph.edge_index = subgraph(select_idx, graph.edge_index,
+                                            relabel_nodes=True)[0]
+            sub_graph.batch = graph.batch[select_idx]
+            sub_graph.ptr = None
+            return feat, sub_graph
+        else:
+            return feat
+
+
+class GraphUnpool(torch.nn.Module):
+    r"""
+    Description
+    -----------
+    The unpooling module for graph
+
+    Parameters
+    ----------
+    hidden_dim : int
+        The number of channels of node features.
+    """
+    def __init__(self, hidden_dim: int):
+        super(GraphUnpool, self).__init__()
+        self.up_sample_gcn = GCNConvWithDropout(hidden_dim, hidden_dim)
+
+    def forward(self, graph: Batch, feat: Tensor, select_idx: Tensor):
+        r"""
+        Description
+        -----------
+        Perform graph unpooling
+
+        Parameters
+        ----------
+        graph : torch_geometric.data.Batch
+            The input graph
+        feat : torch.Tensor
+            The input node feature
+        select_idx : torch.Tensor
+            The index in fine graph of node from
+            coarse graph, this is obtained from
+            previous graph pooling layers.
+        """
+        fine_feat = torch.zeros((graph.num_nodes, feat.size(-1)),
+                                device=feat.device)
+        fine_feat[select_idx] = feat
+        fine_feat = self.up_sample_gcn(fine_feat, graph.edge_index)
+        return fine_feat
+
+
+class GraphCrossModule(torch.nn.Module):
+    r"""
```

Comment:
Please follow the docstring conventions of our other modules to ensure that documentation is rendered correctly, see [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/mlp.py#L11-L24).